File: bank-kyc-system/src/main/java/com/bank/project/entity/BankAccount.java
package com.bank.project.entity;

import java.math.BigDecimal;
import java.util.List;

import com.bank.project.enums.AccountStatus;
import com.bank.project.enums.AccountType;
import com.bank.project.enums.KycStatus;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class BankAccount {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String accountNumber;

    @Enumerated(EnumType.STRING)
    private AccountType accountType;

    @Column
    private String branchName;

    @Column
    private String ifscCode;

    @Column
    private BigDecimal balance;

    @Enumerated(EnumType.STRING)
    @Column
    private AccountStatus status;
    
    @Enumerated(EnumType.STRING)
    
    private KycStatus kycStatus = KycStatus.PENDING;

    @OneToMany(mappedBy = "account", cascade = CascadeType.ALL)
    private List<Transaction> transactions;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;
}

File: BankKycSystemdotnet/BankKycSystem/Entities/BankAccount.cs

using BankKycSystem.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Numerics;
// Assuming your enums are here


namespace BankKycSystem.Entities
{
    public class BankAccount
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long? Id { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string? AccountNumber { get; set; }

        public AccountType? AccountType { get; set; }

        [Column(TypeName = "nvarchar(100)")]
        public string? BranchName { get; set; }

        [Column(TypeName = "nvarchar(50)")]
        public string? IfscCode { get; set; }

        [Column(TypeName = "decimal(18,2)")]
        public decimal? Balance { get; set; }

        public AccountStatus? Status { get; set; }

        public KycStatus? KycStatus { get; set; } = BankKycSystem.Enums.KycStatus.PENDING;

        
        public virtual List<Transaction>? Transactions { get; set; }

        [ForeignKey("UserId")]
        public long? UserId { get; set; }

        public virtual User? User { get; set; }
    }
}
